test/test_unary_ufuncs.py

```diff
@@ -33,6 +202,190 @@ def test_unsupported_dtypes(self, device, dtype, op):
         with self.assertRaises(RuntimeError):
             op(t)
 
+    # Tests bool tensor negation raises the correct error
+    def test_neg_error_message(self, device):
+        self.assertRaisesRegex(
+            RuntimeError,
+            r"Negation, the `\-` operator, on a bool tensor is not supported. "
+            r"If you are trying to invert a mask, use the `\~` or `logical_not\(\)` operator instead.",
+            lambda: - torch.tensor([False, True], device=device))
+
+    @dtypes(*floating_types_and(torch.bfloat16, torch.half))
+    @ops((_fn for _fn in unary_ufuncs if _fn.domain != (None, None)))
+    def test_float_domains(self, device, dtype, op):
+        if not op.supports_dtype(dtype, torch.device(device).type):
+            raise unittest.SkipTest('unsupported dtype')
+
+        eps = (1e-1, 1, 2, 10, 20, 50, 100)
+
+        # Adds smaller values except on bfloat16, since these small differences
+        #   are often too small to represent in bfloat16
+        if dtype is not torch.bfloat16:
+            eps = eps + (1e-5, 1e-3)
+
```

Comment:
low, high = op.domain

Reply:
Yeah, this seems clearer to me. Also consider making op.domain a named tuple; will support destructuring and named access too.

Reply:
Let's revisit domain representation in a follow-up PR.